src/core-services/settings/mutations/updateAppSettings.js

```diff
@@ -25,18 +25,22 @@ export default async function updateAppSettings(context, settingsUpdates, shopId
 
   // Look up roles that are allowed to set each setting. Throw if not allowed.
   for (const updateKey of updateKeys) {
-    const allowedRoles = shopId ? rolesThatCanEditShopSetting(updateKey) : rolesThatCanEditGlobalSetting(updateKey);
+    const allowedRoles = shopId ? permissionsThatCanEditShopSetting(updateKey) : permissionsThatCanEditGlobalSetting(updateKey);
```

Comment:
`allowedRoles` could also be renamed to `permissionsNeededToEdit` to match other renaming